### PR TITLE
MNT: Use hash for Action workflow versions and update if needed

### PR DESCRIPTION
**Pull request by pllim**
As recommended by https://scientific-python.org/specs/spec-0008/#pin-github-actions-release-workflows-to-their-full-release-commit-shas , this PR changes your Actions workflow version pins to hashes, and updates to latest release hashes (at the time of writing) if needed.

This is an automated update made by the `batchpr` tool :robot: - feel free to close if it doesn't look good! You can report issues to @pllim.

[:ghost:](https://github.com/pllim/playpen/issues/60)

### CODE REVIEW

**Comment by Cadair** [comment]
@pllim do we need to do anything to configure dependabot to handle this format?

**Comment by pllim** [comment]
No, I don't think so. @bsipocz can confirm if you need second opinion. Thanks!

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -217,23 +217,23 @@ jobs:
...
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938  # v4.2.0
         with:
           fetch-depth: 0
           lfs: true
           submodules: ${{ inputs.submodules }}
       - name: Install dependencies
         if: ${{ inputs.libraries != '' }}
-        uses: ConorMacBride/install-package@main
+        uses: ConorMacBride/install-package@3e7ad059e07782ee54fa35f827df52aae0626f30  # v1.1.0
```
Comment (bsipocz) [inline .github/workflows/publish.yml:227]: This doesn't seem good to use a personal action in here.

Comment (pllim) [reply .github/workflows/publish.yml:227]: That is a separate issue:

* https://github.com/ConorMacBride/install-package/issues/3

**Review by Cadair — APPROVED** [review]
I am fine with this, but I will let someone else review it as well.

**Comment by pllim** [comment]
> grouped, monthly run

If Stuart wants that, I can also attach that change in this PR real quick. Just lemme know.

**Comment by Cadair** [comment]
Go for it!

**Comment by pllim** [comment]
Done. Thanks, all!

**Comment by Cadair** [comment]
@pllim can you rebase this?

**Comment by pllim** [comment]
@Cadair I used the GitHub web interface to resolve conflicts which introduced merge commit. Please **squash and merge**. Thanks!

**Comment by pllim** [comment]
There are some runner errors/warnings that are not related to this PR?

https://github.com/OpenAstronomy/github-actions-workflows/actions/runs/12015369295

File: .github/workflows/publish_pure_python.yml [hunk 2]
```diff
@@ -134,14 +134,14 @@ jobs:
...
-      - uses: pypa/gh-action-pypi-publish@release/v1
+      - uses: pypa/gh-action-pypi-publish@897895f1e160c830e369f9779632ebc134688e1b  # v1.10.2
         name: Upload to PyPI
         if: ${{ steps.set-upload.outputs.upload_to_pypi == 'true' }}
         with:
           user: __token__
           password: ${{ secrets.pypi_token }}
           repository-url: ${{ inputs.repository_url }}
-      - uses: OpenAstronomy/publish-wheels-anaconda@main
+      - uses: OpenAstronomy/publish-wheels-anaconda@612ea808f79152bd52a019316f684a12bbe8ba33  # main
```
Comment (bsipocz) [inline .github/workflows/publish_pure_python.yml:144]: unrelated but there should be versions tagged in this repo, too.

Comment (bsipocz) [reply .github/workflows/publish_pure_python.yml:144]: now I see that you have already requested it.

**Comment by bsipocz** [comment]
Thanks @pllim!

**Pull request merged**